<div class="alert alert-block alert-info">
<center>ESTATÍSTICA APLICADA</center>
<center>PROJETO 01</center>
<center>ALUNO: Caio Lívio Leite Muniz Dantas</center>
</div>

In [157]:
import numpy as np
import pandas as pd
import zipfile

<div class="alert alert-block alert-info">
1º Tarefa
</div>

In [158]:
# Pegando os dados do site da PRF

dados_2018 = pd.read_csv('http://arquivos.prf.gov.br/arquivos/index.php/s/MaC6cieXSFACNWT/download', encoding = 'latin1', compression = 'zip', sep = ';')

dados_2019 = pd.read_csv('http://arquivos.prf.gov.br/arquivos/index.php/s/kRBUylqz6DyQznN/download', encoding = 'latin1', compression = 'zip', sep = ';')

dados_2020 = pd.read_csv('http://arquivos.prf.gov.br/arquivos/index.php/s/jdDLrQIf33xXSCe/download', encoding = 'latin1', compression = 'zip', sep = ';')

dados_2021 = pd.read_csv('http://arquivos.prf.gov.br/arquivos/index.php/s/n1T3lymvIdDOzzb/download', encoding = 'latin1', compression = 'zip', sep = ';')


- Peguei todos os dados do site da PRF separados em anos de 2019 até 2021, sem precisar baixar os arquivor e sim puxando diretamente do site, existe um jeito de fazer isso e foi exatamente o que fiz em cima, pois usei a função chamada "read_csv".


In [159]:
# Todos os dados de 2018 

dados_2018

id data_inversa     dia_semana   horario  uf     br     km  \
0       99973.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   
1       99976.0   2018-01-01  segunda-feira  00:40:00  SC  282.0    0,4   
2       99977.0   2018-01-01  segunda-feira  00:30:00  RJ  493.0      1   
3       99981.0   2018-01-01  segunda-feira  01:15:00  RS  386.0    134   
4       99982.0   2018-01-01  segunda-feira  00:20:00  RS  293.0  151,7   
...         ...          ...            ...       ...  ..    ...    ...   
69290  252256.0   2018-12-24  segunda-feira  09:15:00  PI  343.0  263,1   
69291  252652.0   2018-12-12   quarta-feira  07:00:00  BA  101.0    321   
69292  252801.0   2018-11-15   quinta-feira  10:48:00  SP  116.0    529   
69293  253040.0   2018-11-23    sexta-feira  19:30:00  RS  472.0  573,5   
69294  253078.0   2018-10-01  segunda-feira  04:31:00  GO   40.0      9   

                       municipio  \
0                        RESENDE   
1                  FLORIANOPOLIS   
2                       ITABORAI   
3                        SARANDI   
4                       CANDIOTA   
...                          ...   
69290                CAMPO MAIOR   
69291  PRESIDENTE TANCREDO NEVES   
69292             BARRA DO TURVO   
69293                 URUGUAIANA   
69294                   LUZIANIA   

                                          causa_acidente  \
0                                      Condutor Dormindo   
1                     Não guardar distância de segurança   
2                                 Ultrapassagem Indevida   
3                                     Ingestão de Álcool   
4                            Falta de Atenção à Condução   
...                                                  ...   
69290                                   Agressão Externa   
69291                        Falta de Atenção à Condução   
69292                        Falta de Atenção à Condução   
69293  Desobediência às normas de trânsito pelo condutor   
69294                 Não guardar distância de segurança   

                      tipo_acidente  ... feridos_graves ilesos ignorados  \
0         Saída de leito carroçável  ...              4      1         0   
1                  Colisão traseira  ...              2      1         0   
2                   Colisão frontal  ...              0      3         1   
3               Colisão transversal  ...              0      2         0   
4         Saída de leito carroçável  ...              1      0         0   
...                             ...  ...            ...    ...       ...   
69290   Colisão com objeto estático  ...              0      1         0   
69291     Saída de leito carroçável  ...              0      0         0   
69292              Colisão traseira  ...              0      0         1   
69293  Queda de ocupante de veículo  ...              1      1         0   
69294     Saída de leito carroçável  ...              0      0         0   

      feridos veiculos      latitude     longitude  regional  delegacia  \
0           4        1     -22,46937     -44,44705     SR-RJ     DEL5/7   
1           2        2  -27,59971687  -48,57565694     SR-SC     DEL8/1   
2           1        3    -22,763901    -42,927532     SR-RJ     DEL5/4   
3           0        2    -27,953636    -52,916374     SR-RS    DEL9/14   
4           1        1    -31,395214  -53,78391174     SR-RS    DEL9/11   
...       ...      ...           ...           ...       ...        ...   
69290       0        1     -4,819326    -42,158821     SR-PI    DEL17/1   
69291       1        1  -13,44768518  -39,42066193     SR-BA    DEL10/5   
69292       1        2    -24,939211    -48,301248     SR-SP     DEL6/5   
69293       1        2  -29,76355779  -57,05200195     SR-RS    DEL9/13   
69294       1        1    -16,131158    -47,957739     SR-DF   UOP01/DF   

            uop  
0      UOP03/RJ  
1      UOP01/SC  
2      UOP02/RJ  
3      UOP01/RS  
4      UOP03/RS  
...         ...  
69290  UOP03/PI  


In [160]:
# Todos os dados de 2019

dados_2019

id data_inversa     dia_semana   horario  uf     br     km  \
0      182210.0   2019-01-01    terça-feira  01:30:00  SP  116.0    218   
1      182211.0   2019-01-01    terça-feira  01:30:00  PR  373.0  177,3   
2      182212.0   2019-01-01    terça-feira  00:45:00  SC  101.0     16   
3      182214.0   2019-01-01    terça-feira  00:00:00  CE   20.0  429,5   
4      182215.0   2019-01-01    terça-feira  01:00:00  MG  381.0    327   
...         ...          ...            ...       ...  ..    ...    ...   
67441  266255.0   2019-10-07  segunda-feira  09:10:00  ES  101.0  270,5   
67442  266406.0   2019-11-25  segunda-feira  07:20:00  PR  116.0    125   
67443  266434.0   2019-06-03  segunda-feira  19:00:00  CE  222.0    1,4   
67444  266573.0   2019-07-13         sábado  19:35:00  PR  373.0    425   
67445  266627.0   2019-10-22    terça-feira  20:20:00  BA  324.0    608   

                municipio                       causa_acidente  \
0               GUARULHOS          Falta de Atenção à Condução   
1            PONTA GROSSA          Falta de Atenção à Condução   
2                  GARUVA                     Animais na Pista   
3               FORTALEZA  Ingestão de Substâncias Psicoativas   
4                NOVA ERA          Falta de Atenção à Condução   
...                   ...                                  ...   
67441               SERRA          Falta de Atenção à Condução   
67442  FAZENDA RIO GRANDE          Falta de Atenção à Condução   
67443             CAUCAIA   Não guardar distância de segurança   
67444              CANDOI          Defeito Mecânico no Veículo   
67445        SIMOES FILHO                       Defeito na Via   

                      tipo_acidente  ... feridos_graves ilesos ignorados  \
0       Colisão com objeto estático  ...              0      0         0   
1                  Colisão traseira  ...              0      0         0   
2       Colisão com objeto estático  ...              0      0         0   
3       Colisão com objeto estático  ...              1      0         0   
4               Colisão transversal  ...              1      1         0   
...                             ...  ...            ...    ...       ...   
67441                    Tombamento  ...              1      1         0   
67442               Colisão lateral  ...              0      1         1   
67443              Colisão traseira  ...              1      1         0   
67444     Saída de leito carroçável  ...              1      0         0   
67445  Queda de ocupante de veículo  ...              0      0         0   

      feridos veiculos      latitude     longitude  regional  delegacia  \
0           4        1  -23,46052014  -46,48772478     SR-SP     DEL6/1   
1           1        1  -25,05533957  -50,22776753     SR-PR     DEL7/3   
2           1        1      -26,1216      -48,8826     SR-SC     DEL8/3   
3           1        1      -3,76999    -38,670063     SR-CE    DEL16/1   
4           1        2      -19,7609      -43,0306     SR-MG     DEL4/3   
...       ...      ...           ...           ...       ...        ...   
67441       1        2     -20,23107     -40,27387     SR-ES    DEL12/2   
67442       1        2     -25,59505     -49,31631     SR-PR     DEL7/1   
67443       1        2     -3,736507     -38,65337     SR-CE    DEL16/1   
67444       1        1      -25,7118     -52,18374     SR-PR     DEL7/3   
67445       1        1     -12,78156     -38,41656     SR-BA    DEL10/1   

            uop  
0      UOP01/SP  
1      UOP01/PR  
2      UOP01/SC  
3      UOP02/CE  
4      UOP01/MG  
...         ...  
67441  UOP01/ES  
67442  UOP03/PR  
67443  UOP01/CE  
67444  UOP02/PR  
67445  UOP01/BA  

[67446 rows x 30 columns]

In [161]:
# Todos os dados de 2020

dados_2020

id data_inversa    dia_semana   horario  uf     br     km  \
0      260031.0   2020-01-01  quarta-feira  01:00:00  TO  153.0  678,1   
1      260036.0   2020-01-01  quarta-feira  01:00:00  RJ  116.0    178   
2      260037.0   2020-01-01  quarta-feira  01:52:00  SC  101.0  206,9   
3      260038.0   2020-01-01  quarta-feira  01:15:00  RO  364.0    236   
4      260039.0   2020-01-01  quarta-feira  01:30:00  PR  153.0    360   
...         ...          ...           ...       ...  ..    ...    ...   
63543  364615.0   2020-03-20   sexta-feira  09:00:00  RN  304.0  171,8   
63544  364810.0   2020-09-12        sábado  23:50:00  BA  116.0   11,5   
63545  364817.0   2020-10-30   sexta-feira  10:30:00  BA  116.0   14,3   
63546  364835.0   2020-09-20       domingo  23:00:00  BA  116.0      8   
63547  364903.0   2020-10-31        sábado  02:10:00  BA  116.0      3   

                  municipio                causa_acidente  \
0                    GURUPI              Animais na Pista   
1               NOVA IGUACU  Falta de Atenção do Pedestre   
2                  SAO JOSE            Ingestão de Álcool   
3                    CACOAL       Velocidade Incompatível   
4                  REBOUCAS            Ingestão de Álcool   
...                     ...                           ...   
63543               ANGICOS   Falta de Atenção à Condução   
63544  VITORIA DA CONQUISTA            Ingestão de Álcool   
63545  VITORIA DA CONQUISTA   Falta de Atenção à Condução   
63546  VITORIA DA CONQUISTA       Velocidade Incompatível   
63547  VITORIA DA CONQUISTA              Animais na Pista   

                      tipo_acidente  ... feridos_graves ilesos ignorados  \
0           Atropelamento de Animal  ...              5      0         0   
1         Atropelamento de Pedestre  ...              0      1         0   
2                   Colisão lateral  ...              0      1         0   
3         Saída de leito carroçável  ...              1      0         0   
4                       Capotamento  ...              4      0         0   
...                             ...  ...            ...    ...       ...   
63543              Colisão traseira  ...              1      2         0   
63544               Colisão lateral  ...              0      2         0   
63545               Colisão lateral  ...              0      2         0   
63546  Queda de ocupante de veículo  ...              0      0         0   
63547       Atropelamento de Animal  ...              1      0         0   

      feridos veiculos      latitude     longitude  regional  delegacia  \
0           5        1  -11,77460203  -49,10744996     SR-TO   UOP01/TO   
1           0        1  -22,75223028   -43,4379103     SR-RJ     DEL5/1   
2           1        2  -27,59193546  -48,61824557     SR-SC     DEL8/1   
3           1        1  -11,44624577  -61,43761218     SR-RO    DEL21/2   
4           6        1  -25,67503796  -50,75089805     SR-PR     DEL7/3   
...       ...      ...           ...           ...       ...        ...   
63543       2        4   -5,68711053    -36,419458     SR-RN      UOP03   
63544       1        2  -14,87308214  -40,87141514     SR-BA      UOP01   
63545       2        2  -14,89597653  -40,86098671     SR-BA      UOP01   
63546       1        1  -14,84732305  -40,88634968     SR-BA      UOP01   
63547       1        1    -14,824848    -40,861311     SR-BA      UOP01   

            uop  
0           NaN  
1      UOP02/RJ  
2      UOP01/SC  
3      UOP02/RO  
4      UOP04/PR  
...         ...  
63543     DEL01  
63544     DEL08  
63545     DEL08  
63546     DEL08  
63547     DEL08  

[63548 rows x 30 columns]

In [162]:
# Todos os dados de 2021

dados_2021

id data_inversa     dia_semana   horario  uf     br     km  \
0      331693.0   2021-01-01    sexta-feira  00:02:00  SC  101.0  314,9   
1      331694.0   2021-01-01    sexta-feira  00:15:00  PR  376.0    188   
2      331696.0   2021-01-01    sexta-feira  00:00:00  SP  116.0  110,6   
3      331699.0   2021-01-01    sexta-feira  01:20:00  SC  101.0    111   
4      331701.0   2021-01-01    sexta-feira  00:30:00  PE  101.0   45,1   
...         ...          ...            ...       ...  ..    ...    ...   
64510  425426.0   2021-05-13   quinta-feira  03:15:00  MG  381.0  775,3   
64511  425436.0   2021-11-15  segunda-feira  23:00:00  PA   10.0  248,4   
64512  428107.0   2021-09-21    terça-feira  01:10:00  MG   50.0     20   
64513  435016.0   2021-12-29   quarta-feira  13:40:00  AC  364.0    170   
64514  441173.0   2021-08-22        domingo  17:10:00  MT  364.0    498   

            municipio                                     causa_acidente  \
0              LAGUNA                   Ingestão de álcool pelo condutor   
1            MARIALVA                            Velocidade Incompatível   
2             TAUBATE               Demais falhas mecânicas ou elétricas   
3          NAVEGANTES                        Manobra de mudança de faixa   
4            IGARASSU           Reação tardia ou ineficiente do condutor   
...               ...                                                ...   
64510        CAMPANHA  Condutor deixou de manter distância do veículo...   
64511  AURORA DO PARA                             Transitar na contramão   
64512        ARAGUARI                                  Condutor Dormindo   
64513          BUJARI                                   Pista esburacada   
64514         JANGADA           Reação tardia ou ineficiente do condutor   

                      tipo_acidente  ... feridos_graves ilesos ignorados  \
0                  Colisão traseira  ...              0      2         0   
1                Colisão com objeto  ...              0      0         0   
2                          Incêndio  ...              0      1         0   
3                   Colisão lateral  ...              0      1         0   
4                  Colisão traseira  ...              1      1         0   
...                             ...  ...            ...    ...       ...   
64510              Colisão traseira  ...              0      2         0   
64511           Colisão transversal  ...              1      1         0   
64512            Colisão com objeto  ...              2      1         0   
64513  Queda de ocupante de veículo  ...              1      0         0   
64514              Colisão traseira  ...              0      1         0   

      feridos veiculos   latitude  longitude  regional  delegacia    uop  
0           1        2 -28.430294 -48.849251     SR-SC      UOP01  DEL02  
1           2        1 -23.466349 -51.828080     SR-PR      UOP01  DEL09  
2           0        1 -23.034126 -45.552700     SR-SP      UOP01  DEL06  
3           2        2 -26.839557 -48.714181     SR-SC      UOP04  DEL04  
4           1        2  -7.865213 -34.907152     SR-PE      UOP02  DEL01  
...       ...      ...        ...        ...       ...        ...    ...  
64510       2        2 -21.790183 -45.446659     SR-MG      UOP02  DEL16  
64511       2        2  -2.267754 -47.563084     SR-PA      UOP01  DEL02  
64512       7        1 -18.587210 -48.056456     SR-MG      UOP02  DEL15  
64513       1        1  -9.766854 -68.056811     SR-AC      UOP01  DEL01  
64514       1        2 -15.263832 -56.455487     SR-MT      UOP02  DEL01  

[64515 rows x 30 columns]

In [164]:
# Juntandos todos os dados de todos os anos

todos_os_anos = pd.concat([dados_2018, dados_2019, dados_2020, dados_2021], keys = ['2018','2019','2020','2021'])
todos_os_anos

id data_inversa     dia_semana   horario  uf     br     km  \
2018 0       99973.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   
     1       99976.0   2018-01-01  segunda-feira  00:40:00  SC  282.0    0,4   
     2       99977.0   2018-01-01  segunda-feira  00:30:00  RJ  493.0      1   
     3       99981.0   2018-01-01  segunda-feira  01:15:00  RS  386.0    134   
     4       99982.0   2018-01-01  segunda-feira  00:20:00  RS  293.0  151,7   
...              ...          ...            ...       ...  ..    ...    ...   
2021 64510  425426.0   2021-05-13   quinta-feira  03:15:00  MG  381.0  775,3   
     64511  425436.0   2021-11-15  segunda-feira  23:00:00  PA   10.0  248,4   
     64512  428107.0   2021-09-21    terça-feira  01:10:00  MG   50.0     20   
     64513  435016.0   2021-12-29   quarta-feira  13:40:00  AC  364.0    170   
     64514  441173.0   2021-08-22        domingo  17:10:00  MT  364.0    498   

                 municipio                                     causa_acidente  \
2018 0             RESENDE                                  Condutor Dormindo   
     1       FLORIANOPOLIS                 Não guardar distância de segurança   
     2            ITABORAI                             Ultrapassagem Indevida   
     3             SARANDI                                 Ingestão de Álcool   
     4            CANDIOTA                        Falta de Atenção à Condução   
...                    ...                                                ...   
2021 64510        CAMPANHA  Condutor deixou de manter distância do veículo...   
     64511  AURORA DO PARA                             Transitar na contramão   
     64512        ARAGUARI                                  Condutor Dormindo   
     64513          BUJARI                                   Pista esburacada   
     64514         JANGADA           Reação tardia ou ineficiente do condutor   

                           tipo_acidente  ... feridos_graves ilesos ignorados  \
2018 0         Saída de leito carroçável  ...              4      1         0   
     1                  Colisão traseira  ...              2      1         0   
     2                   Colisão frontal  ...              0      3         1   
     3               Colisão transversal  ...              0      2         0   
     4         Saída de leito carroçável  ...              1      0         0   
...                                  ...  ...            ...    ...       ...   
2021 64510              Colisão traseira  ...              0      2         0   
     64511           Colisão transversal  ...              1      1         0   
     64512            Colisão com objeto  ...              2      1         0   
     64513  Queda de ocupante de veículo  ...              1      0         0   
     64514              Colisão traseira  ...              0      1         0   

           feridos veiculos      latitude     longitude  regional  delegacia  \
2018 0           4        1     -22,46937     -44,44705     SR-RJ     DEL5/7   
     1           2        2  -27,59971687  -48,57565694     SR-SC     DEL8/1   
     2           1        3    -22,763901    -42,927532     SR-RJ     DEL5/4   
     3           0        2    -27,953636    -52,916374     SR-RS    DEL9/14   
     4           1        1    -31,395214  -53,78391174     SR-RS    DEL9/11   
...            ...      ...           ...           ...       ...        ...   
2021 64510       2        2    -21.790183    -45.446659     SR-MG      UOP02   
     64511       2        2     -2.267754    -47.563084     SR-PA      UOP01   
     64512       7        1     -18.58721    -48.056456     SR-MG      UOP02   
     64513       1        1     -9.766854    -68.056811     SR-AC      UOP01   
     64514       1        2    -15.263832    -56.455487     SR-MT      UOP02   

                 uop  
2018 0      UOP03/RJ  
     1      UOP01/SC  
     2      UOP02/RJ  
     3      UOP01/RS  
     4      UOP03/RS  
...         

- Nesse caso, juntei todos os dados de todos os anos usando a função "concat".

<div class="alert alert-block alert-info">
2º Tarefa - A
</div>

In [165]:
len(dados_2018)

69295

Ocorreram 69295 acidentes em 2018.

In [166]:
len(dados_2019)

67446

Ocorreram 67446 acidentes em 2019.

In [167]:
len(dados_2020)

63548

Ocorreram 63548 acidentes em 2020.

In [168]:
len(dados_2021)

64515

Ocorreram 64515 acidentes em 2021.

- Usando a função len() conseguimos observar o número de index e assim saber quantos acidentes ocorreram em cada ano.

<div class="alert alert-block alert-info">
2º Tarefa - B
</div>

In [169]:
dados_2018.shape

(69295, 30)

In [170]:
dados_2019.shape

(67446, 30)

In [171]:
dados_2020.shape

(63548, 30)

In [172]:
dados_2021.shape

(64515, 30)

- Como se pode observar a função "shape" indica a quantidade de colunas e linhas que o dataframe armazena, e para saber a quantidade de variáveis que um dataframe possui é só observar a quantidade de colunas, assim todos os conjuntos de dados registra 30 variáveis.

<div class="alert alert-block alert-info">
2º Tarefa - C
</div>

In [173]:
cid = pd.DataFrame(todos_os_anos['municipio'].value_counts()).iloc[:5,:1]
cid = cid.reset_index().rename(columns = {'index':'Municipios','municipio':'Número de Acidentes'})
cid

Municipios  Número de Acidentes
0   BRASILIA                 3978
1   CURITIBA                 3962
2   SAO JOSE                 3011
3  GUARULHOS                 2781
4    PALHOCA                 2496

- Basicamente, criei um dataframe recebendo somente os valores relacionados aos municípios e na mesma linha usei a função "value_counts" que retorna o valor da contagem exibindo em ordem decrescente para que o elemento mais frequente venha primeiro e assim usei também a função "iloc" que limita a quantidade de valores dentro de uma coluna, e como eu quero somente as 5 primeiras cidades coloquei [:5,:1]. Assim, na outra linha, basicamente organizei os nomes das columas usando "rename".

<div class="alert alert-block alert-info">
2º Tarefa - D
</div>

In [174]:
pb = dados_2019[(dados_2019.uf == "PB")]
pb

id data_inversa     dia_semana   horario  uf     br     km  \
16     182232.0   2019-01-01    terça-feira  03:00:00  PB  230.0  501,3   
47     182271.0   2019-01-01    terça-feira  04:15:00  PB  361.0   38,8   
82     182312.0   2019-01-01    terça-feira  13:45:00  PB  230.0   18,9   
220    182480.0   2019-01-02   quarta-feira  07:30:00  PB  230.0     57   
293    182577.0   2019-01-02   quarta-feira  16:30:00  PB  104.0    100   
...         ...          ...            ...       ...  ..    ...    ...   
66954  259678.0   2019-12-30  segunda-feira  15:35:00  PB  230.0    505   
67009  259737.0   2019-12-30  segunda-feira  19:00:00  PB  230.0  378,7   
67050  259780.0   2019-12-31    terça-feira  00:40:00  PB  101.0    122   
67100  259833.0   2019-12-31    terça-feira  12:15:00  PB  101.0   41,9   
67208  260062.0   2019-12-31    terça-feira  23:40:00  PB  434.0    5,7   

                             municipio  \
16                          CAJAZEIRAS   
47                         CATINGUEIRA   
82                         JOAO PESSOA   
220             CRUZ DO ESPIRITO SANTO   
293    SAO SEBASTIAO DE LAGOA DE  ROCA   
...                                ...   
66954                       CAJAZEIRAS   
67009                          CONDADO   
67050                          CAAPORA   
67100                       MAMANGUAPE   
67208                          UIRAUNA   

                                          causa_acidente  \
16                           Falta de Atenção à Condução   
47                                    Ingestão de Álcool   
82                           Defeito Mecânico no Veículo   
220                              Velocidade Incompatível   
293                          Falta de Atenção à Condução   
...                                                  ...   
66954                                         Mal Súbito   
67009  Desobediência às normas de trânsito pelo pedestre   
67050                            Velocidade Incompatível   
67100                                     Defeito na Via   
67208                            Velocidade Incompatível   

                      tipo_acidente  ... feridos_graves ilesos ignorados  \
16                  Colisão frontal  ...              0      0         0   
47        Saída de leito carroçável  ...              1      0         0   
82                       Tombamento  ...              1      1         0   
220                Colisão traseira  ...              0      1         0   
293                Colisão traseira  ...              0      2         0   
...                             ...  ...            ...    ...       ...   
66954  Queda de ocupante de veículo  ...              1      0         0   
67009     Atropelamento de Pedestre  ...              0      4         0   
67050              Colisão traseira  ...              1      1         0   
67100   Colisão com objeto estático  ...              0      2         1   
67208                    Tombamento  ...              0      0         0   

      feridos veiculos     latitude     longitude  regional  delegacia  \
16          1        2       -6,899      -38,5159     SR-PB    DEL14/3   
47          1        1  -7,10157405  -37,60036469     SR-PB    DEL14/3   
82          1        2    -7,129088    -34,850031     SR-PB    DEL14/1   
220         1        2      -7,2028      -35,1125     SR-PB    DEL14/1   
293         0        2    -7,045145    -35,857178     SR-PB    DEL14/2   
...       ...      ...          ...           ...       ...        ...   
66954       1        1      -6,8995      -38,5494     SR-PB    DEL14/3   
67009       0        1      -6,9134      -37,6178     SR-PB    DEL14/3   
67050       1        2    -7,450382     -34,97994     SR-PB    DEL14/1   
67100       0        2    -6,847666      -35,1355     SR-PB    DEL14/1   
67208       0        1    -6,464712     -38,42677     SR-PB    DEL14/3   

            uop  
16     UOP03/PB  
47     UOP01/PB  
82     UOP00/PB  
220    UOP02

In [175]:
pb[pb.feridos_graves > 0].feridos_graves.count()

414

Tiveram 414 acidentes com feridos graves aconteceram na Paraíba em 2019.

- Primeiramente, eu filtrei somente com dados de 2019 no estado da paraíba, depois peguei a coluna de meu interresse (feridos_graves) e coloquei para somar todos os valores acima de 0.

<div class="alert alert-block alert-info">
3º Tarefa - A
</div>

In [177]:
# Ano de 2018

est = dados_2018["uf"].value_counts()
est = est.reset_index().rename(columns = {"index":"UF", "uf":"Número de Acidentes"})
est.to_csv('acidentes_por_estado.csv')
est

UF  Número de Acidentes
0   MG                 9066
1   SC                 8497
2   PR                 7952
3   RJ                 4575
4   SP                 4516
5   RS                 4445
6   BA                 3555
7   GO                 3464
8   PE                 2715
9   ES                 2642
10  MT                 2631
11  CE                 1697
12  RO                 1629
13  PB                 1588
14  MS                 1530
15  RN                 1416
16  PI                 1371
17  MA                 1272
18  PA                 1260
19  DF                  867
20  AL                  691
21  TO                  589
22  SE                  559
23  RR                  236
24  AC                  235
25  AP                  173
26  AM                  124

In [178]:
# Ano de 2019

est = dados_2019["uf"].value_counts()
est = est.reset_index().rename(columns = {"index":"UF", "uf":"Número de Acidentes"})
est.to_csv('acidentes_por_estado.csv')
est

UF  Número de Acidentes
0   MG                 8720
1   SC                 8428
2   PR                 7708
3   RJ                 4606
4   RS                 4595
5   SP                 4377
6   BA                 3471
7   GO                 3354
8   ES                 2842
9   PE                 2736
10  MT                 2137
11  CE                 1694
12  PB                 1564
13  RO                 1533
14  RN                 1437
15  MS                 1435
16  PI                 1346
17  DF                 1090
18  MA                 1075
19  PA                  817
20  AL                  629
21  SE                  612
22  TO                  520
23  AC                  254
24  RR                  237
25  AP                  128
26  AM                  101

In [179]:
# Ano de 2020

est = dados_2020["uf"].value_counts()
est = est.reset_index().rename(columns = {"index":"UF", "uf":"Número de Acidentes"})
est.to_csv('acidentes_por_estado.csv')
est

UF  Número de Acidentes
0   MG                 8374
1   SC                 7220
2   PR                 7190
3   RJ                 4236
4   RS                 4176
5   SP                 4043
6   BA                 3456
7   GO                 3228
8   PE                 2560
9   ES                 2534
10  MT                 2195
11  CE                 1571
12  MS                 1540
13  RO                 1486
14  PB                 1401
15  RN                 1337
16  PI                 1203
17  MA                 1154
18  DF                 1040
19  PA                  963
20  AL                  628
21  TO                  595
22  SE                  575
23  AC                  300
24  RR                  250
25  AP                  155
26  AM                  138

In [180]:
# Ano de 2021

est = dados_2021["uf"].value_counts()
est = est.reset_index().rename(columns = {"index":"UF", "uf":"Número de Acidentes"})
est.to_csv('acidentes_por_estado.csv')
est

UF  Número de Acidentes
0   MG                 8323
1   SC                 7889
2   PR                 7335
3   RJ                 4565
4   RS                 4538
5   SP                 4300
6   BA                 3311
7   GO                 3187
8   PE                 2555
9   ES                 2525
10  MT                 2138
11  MS                 1642
12  CE                 1543
13  PB                 1440
14  RO                 1419
15  RN                 1337
16  PI                 1137
17  MA                 1130
18  DF                  981
19  PA                  844
20  AL                  580
21  SE                  509
22  TO                  501
23  RR                  268
24  AC                  255
25  AP                  170
26  AM                   93

In [181]:
# Todos os anos juntos

est = todos_os_anos["uf"].value_counts()
est = est.reset_index().rename(columns = {"index":"UF", "uf":"Número de Acidentes"})
est.to_csv('acidentes_por_estado.csv')
est

UF  Número de Acidentes
0   MG                34483
1   SC                32034
2   PR                30185
3   RJ                17982
4   RS                17754
5   SP                17236
6   BA                13793
7   GO                13233
8   PE                10566
9   ES                10543
10  MT                 9101
11  CE                 6505
12  MS                 6147
13  RO                 6067
14  PB                 5993
15  RN                 5527
16  PI                 5057
17  MA                 4631
18  DF                 3978
19  PA                 3884
20  AL                 2528
21  SE                 2255
22  TO                 2205
23  AC                 1044
24  RR                  991
25  AP                  626
26  AM                  456

- Basicamente, peguei as informações de todos os anos e peguei só a coluna que me interessava, por isso usei o "value_counts" e então, renomei os nomes das colunas para ficar algo mais organizado e finalmente transformei para "csv".

<div class="alert alert-block alert-info">
3º Tarefa - B
</div>

In [182]:
# Ano de 2018

acidentes = pd.DataFrame(dados_2018["dia_semana"].value_counts())
acidentes = acidentes.reset_index().rename(columns = {"dia_semana": "Número de Acidentes", "index": "Dias da Semana"})
acidentes.to_csv('acidentes_por_dia.csv')
acidentes

Dias da Semana  Número de Acidentes
0        domingo                11283
1         sábado                11239
2    sexta-feira                10632
3  segunda-feira                 9422
4   quinta-feira                 9230
5   quarta-feira                 8824
6    terça-feira                 8665

In [183]:
# Ano de 2019

acidentes = pd.DataFrame(dados_2019["dia_semana"].value_counts())
acidentes = acidentes.reset_index().rename(columns = {"dia_semana": "Número de Acidentes", "index": "Dias da Semana"})
acidentes.to_csv('acidentes_por_dia.csv')
acidentes

Dias da Semana  Número de Acidentes
0        domingo                11298
1         sábado                11099
2    sexta-feira                10381
3  segunda-feira                 9134
4   quinta-feira                 8755
5    terça-feira                 8405
6   quarta-feira                 8374

In [184]:
# Ano de 2020

acidentes = pd.DataFrame(dados_2020["dia_semana"].value_counts())
acidentes = acidentes.reset_index().rename(columns = {"dia_semana": "Número de Acidentes", "index": "Dias da Semana"})
acidentes.to_csv('acidentes_por_dia.csv')
acidentes

Dias da Semana  Número de Acidentes
0         sábado                10667
1        domingo                10420
2    sexta-feira                 9764
3  segunda-feira                 8519
4   quinta-feira                 8244
5   quarta-feira                 8155
6    terça-feira                 7779

In [185]:
# Ano de 2021

acidentes = pd.DataFrame(dados_2021["dia_semana"].value_counts())
acidentes = acidentes.reset_index().rename(columns = {"dia_semana": "Número de Acidentes", "index": "Dias da Semana"})
acidentes.to_csv('acidentes_por_dia.csv')
acidentes

Dias da Semana  Número de Acidentes
0         sábado                10867
1        domingo                10815
2    sexta-feira                10168
3  segunda-feira                 8637
4   quinta-feira                 8242
5   quarta-feira                 7994
6    terça-feira                 7792

In [186]:
# Todos os anos juntos

acidentes = pd.DataFrame(todos_os_anos["dia_semana"].value_counts())
acidentes = acidentes.reset_index().rename(columns = {"dia_semana": "Número de Acidentes", "index": "Dias da Semana"})
acidentes.to_csv('acidentes_por_dia.csv')
acidentes

Dias da Semana  Número de Acidentes
0         sábado                43872
1        domingo                43816
2    sexta-feira                40945
3  segunda-feira                35712
4   quinta-feira                34471
5   quarta-feira                33347
6    terça-feira                32641

- Usei a mesma lógica que usei na letra A.

<div class="alert alert-block alert-info">
4º Tarefa - A
</div>

In [187]:
# Causas do ano de 2018

df_2018 = pd.DataFrame(dados_2018["causa_acidente"].value_counts())
df_2018 = df_2018.reset_index().rename(columns = {"causa_acidente": "Quantidade de acidentes", "index": "Causa dos acidentes"})
df_2018 = df_2018.sort_values(by = "Quantidade de acidentes", ascending = False, ignore_index = True)
df_2018

Causa dos acidentes  Quantidade de acidentes
0                         Falta de Atenção à Condução                    25830
1   Desobediência às normas de trânsito pelo condutor                     7189
2                             Velocidade Incompatível                     6849
3                                  Ingestão de Álcool                     5201
4                  Não guardar distância de segurança                     4262
5                         Defeito Mecânico no Veículo                     3716
6                                   Condutor Dormindo                     2633
7                                  Pista Escorregadia                     2549
8                        Falta de Atenção do Pedestre                     2073
9                                    Animais na Pista                     1614
10                             Ultrapassagem Indevida                     1333
11                                     Defeito na Via                     1126
12            Avarias e/ou desgaste excessivo no pneu                     1011
13                                         Mal Súbito                      677
14                          Restrição de Visibilidade                      539
15           Objeto estático sobre o leito carroçável                      539
16             Carga excessiva e/ou mal acondicionada                      367
17                                   Agressão Externa                      356
18                              Fenômenos da Natureza                      310
19  Ingestão de álcool e/ou substâncias psicoativa...                      297
20      Sinalização da via insuficiente ou inadequada                      292
21  Desobediência às normas de trânsito pelo pedestre                      273
22  Deficiência ou não Acionamento do Sistema de I...                      186
23                Ingestão de Substâncias Psicoativas                       73

In [188]:
# Ano de 2019

df_2019 = pd.DataFrame(dados_2019["causa_acidente"].value_counts())
df_2019 = df_2019.reset_index().rename(columns = {"causa_acidente": "Quantidade de acidentes", "index": "Causa dos acidentes"})
df_2019 = df_2019.sort_values(by = "Quantidade de acidentes", ascending = False, ignore_index = True)
df_2019

Causa dos acidentes  Quantidade de acidentes
0                         Falta de Atenção à Condução                    24992
1   Desobediência às normas de trânsito pelo condutor                     8098
2                             Velocidade Incompatível                     6028
3                                  Ingestão de Álcool                     5419
4                  Não guardar distância de segurança                     4214
5                         Defeito Mecânico no Veículo                     3732
6                                   Condutor Dormindo                     2483
7                                  Pista Escorregadia                     2186
8                        Falta de Atenção do Pedestre                     1971
9                                    Animais na Pista                     1527
10                             Ultrapassagem Indevida                     1173
11                                     Defeito na Via                      973
12            Avarias e/ou desgaste excessivo no pneu                      799
13                                         Mal Súbito                      715
14                          Restrição de Visibilidade                      493
15           Objeto estático sobre o leito carroçável                      437
16  Ingestão de álcool e/ou substâncias psicoativa...                      394
17  Desobediência às normas de trânsito pelo pedestre                      392
18                                   Agressão Externa                      384
19             Carga excessiva e/ou mal acondicionada                      305
20      Sinalização da via insuficiente ou inadequada                      247
21                              Fenômenos da Natureza                      230
22  Deficiência ou não Acionamento do Sistema de I...                      160
23                Ingestão de Substâncias Psicoativas                       94

In [189]:
# Ano de 2020

df_2020 = pd.DataFrame(dados_2020["causa_acidente"].value_counts())
df_2020 = df_2020.reset_index().rename(columns = {"causa_acidente": "Quantidade de acidentes", "index": "Causa dos acidentes"})
df_2020 = df_2020.sort_values(by = "Quantidade de acidentes", ascending = False, ignore_index = True)
df_2020

Causa dos acidentes  Quantidade de acidentes
0                         Falta de Atenção à Condução                    22409
1   Desobediência às normas de trânsito pelo condutor                     8802
2                             Velocidade Incompatível                     5886
3                                  Ingestão de Álcool                     5069
4                         Defeito Mecânico no Veículo                     3742
5                  Não guardar distância de segurança                     3606
6                                   Condutor Dormindo                     2109
7                                  Pista Escorregadia                     2017
8                        Falta de Atenção do Pedestre                     1641
9                                    Animais na Pista                     1410
10                             Ultrapassagem Indevida                     1084
11                                     Defeito na Via                      960
12            Avarias e/ou desgaste excessivo no pneu                      902
13                                         Mal Súbito                      584
14                          Restrição de Visibilidade                      483
15           Objeto estático sobre o leito carroçável                      446
16  Desobediência às normas de trânsito pelo pedestre                      388
17             Carga excessiva e/ou mal acondicionada                      377
18  Ingestão de álcool e/ou substâncias psicoativa...                      357
19                                   Agressão Externa                      342
20      Sinalização da via insuficiente ou inadequada                      241
21                              Fenômenos da Natureza                      221
22  Deficiência ou não Acionamento do Sistema de I...                      187
23                Ingestão de Substâncias Psicoativas                       86
24           Reação tardia ou ineficiente do condutor                       24
25                     Ausência de reação do condutor                       21
26  Condutor deixou de manter distância do veículo...                       21
27                        Manobra de mudança de faixa                       19
28                             Transitar na contramão                       14
29  Acessar a via sem observar a presença dos outr...                       13
30               Demais falhas mecânicas ou elétricas                       13
31                   Ingestão de álcool pelo condutor                        8
32           Desrespeitar a preferência no cruzamento                        8
33                                 Conversão proibida                        6
34                                              Chuva                        6
35  Trafegar com motocicleta (ou similar) entre as...                        5
36                           Pedestre andava na pista                        4
37  Condutor desrespeitou a iluminação vermelha do...                        4
38                               Falta de acostamento                        3
39                                   Acesso irregular                        3
40                           Transitar no acostamento                        3
41              Afundamento ou ondulação no pavimento                        2
42                      Entrada inopinada do pedestre                        2
43                              Iluminação deficiente                        2
44                             Mal súbito do condutor                        2
45                               Demais falhas na via                        2
46                  Acumulo de água sobre o pavimento                        2
47     Acumulo de areia ou detritos sobre o pavimento                        1
48                                   Pista esburacada                        1
49              Estacionar ou parar em local proibido                        1
50   

In [190]:
# Ano de 2021

df_2021 = pd.DataFrame(dados_2021["causa_acidente"].value_counts())
df_2021 = df_2021.reset_index().rename(columns = {"causa_acidente": "Quantidade de acidentes", "index": "Causa dos acidentes"})
df_2021 = df_2021.sort_values(by = "Quantidade de acidentes", ascending = False, ignore_index = True)
df_2021

Causa dos acidentes  Quantidade de acidentes
0            Reação tardia ou ineficiente do condutor                     6903
1                             Velocidade Incompatível                     6742
2                      Ausência de reação do condutor                     5601
3   Acessar a via sem observar a presença dos outr...                     5194
4   Condutor deixou de manter distância do veículo...                     5117
..                                                ...                      ...
66                               Semáforo com defeito                        8
67                               Modificação proibida                        8
68                                Participar de racha                        7
69                              Sinalização encoberta                        3
70                    Sistema de drenagem ineficiente                        3

[71 rows x 2 columns]

In [191]:
# Todos os anos juntos

df_todos_anos = pd.DataFrame(todos_os_anos["causa_acidente"].value_counts())
df_todos_anos = df_todos_anos.reset_index().rename(columns = {"causa_acidente": "Quantidade de acidentes", "index": "Causa dos acidentes"})
df_todos_anos = df_todos_anos.sort_values(by = "Quantidade de acidentes", ascending = False, ignore_index = True)
df_todos_anos

Causa dos acidentes  Quantidade de acidentes
0                         Falta de Atenção à Condução                    73231
1                             Velocidade Incompatível                    25505
2   Desobediência às normas de trânsito pelo condutor                    24089
3                                  Ingestão de Álcool                    15689
4                  Não guardar distância de segurança                    12082
..                                                ...                      ...
81                               Semáforo com defeito                        8
82                               Modificação proibida                        8
83                                Participar de racha                        7
84                              Sinalização encoberta                        3
85                    Sistema de drenagem ineficiente                        3

[86 rows x 2 columns]

- Usei o mesmo codido tanto para cada ano como para todos os anos, então, usei a função "value_counts()" para escolher somente a coluna que eu queria, depois usei o "reset_index" para resetar e depois renomeei as colunas com a função "rename", logo depois usei "sort_vaules" para ordernar e escolhi de forma decrecente com a função "ascending = False" e o "ignore_index = True" para iniciar a contagem a partir do número 0.

In [192]:
# Ano de 2018

acidente_mais_freq = df_2018.iloc[:1,:]
acidente_mais_freq

Causa dos acidentes  Quantidade de acidentes
0  Falta de Atenção à Condução                    25830

In [193]:
# Ano de 2018

acidente_mais_rara = df_2018.iloc[[-1]]
acidente_mais_rara

Causa dos acidentes  Quantidade de acidentes
23  Ingestão de Substâncias Psicoativas                       73

- Causa do acidente mais frequente de 2018: Falta de Atenção à Condução\
- Causa do acidente mais raro de 2018: Ingestão de Substâncias Psicoativas

In [194]:
# Ano de 2019

acidente_mais_freq = df_2019.iloc[:1,:]
acidente_mais_freq

Causa dos acidentes  Quantidade de acidentes
0  Falta de Atenção à Condução                    24992

In [195]:
# Ano de 2019

acidente_mais_rara = df_2019.iloc[[-1]]
acidente_mais_rara

Causa dos acidentes  Quantidade de acidentes
23  Ingestão de Substâncias Psicoativas                       94

- Causa do acidente mais frequente de 2019: Falta de Atenção à Condução\
- Causa do acidente mais raro de 2019: Ingestão de Substâncias Psicoativas

In [196]:
# Ano de 2020

acidente_mais_freq = df_2020.iloc[:1,:]
acidente_mais_freq

Causa dos acidentes  Quantidade de acidentes
0  Falta de Atenção à Condução                    22409

In [197]:
# Ano de 2020

acidente_mais_rara = df_2020.iloc[[-1]]
acidente_mais_rara

Causa dos acidentes  Quantidade de acidentes
58     Curva acentuada                        1

- Causa do acidente mais frequente de 2020: Falta de Atenção à Condução\
- Causa do acidente mais raro de 2020: Curva acentuada

In [198]:
# Ano de 2021

acidente_mais_freq = df_2021.iloc[:1,:]
acidente_mais_freq

Causa dos acidentes  Quantidade de acidentes
0  Reação tardia ou ineficiente do condutor                     6903

In [199]:
# Ano de 2021

acidente_mais_rara = df_2021.iloc[[-1]]
acidente_mais_rara

Causa dos acidentes  Quantidade de acidentes
70  Sistema de drenagem ineficiente                        3

- Causa do acidente mais frequente de 2021: Reação tardia ou ineficiente do condutor\
- Causa do acidente mais raro de 2021: Sistema de drenagem ineficiente

In [200]:
# Todos os anos juntos

acidente_mais_freq = df_todos_anos.iloc[:1,:]
acidente_mais_freq

Causa dos acidentes  Quantidade de acidentes
0  Falta de Atenção à Condução                    73231

In [201]:
# Todos os anos juntos

acidente_mais_rara = df_todos_anos.iloc[[-1]]
acidente_mais_rara

Causa dos acidentes  Quantidade de acidentes
85  Sistema de drenagem ineficiente                        3

- Causa do acidente mais frequente de todos os anos: Reação tardia ou ineficiente à condução\
- Causa do acidente mais raro de todos os anos: Sistema de drenagem ineficiente

- Nessa parte, basicamente, usei somente a função "iloc", que diz a quantidade de linhas que eu vou querer.

<div class="alert alert-block alert-info">
4º Tarefa - B
</div>

In [204]:
import datetime

In [205]:
# Dados de 2018

dados_2018['mes'] = pd.to_datetime(dados_2018['data_inversa']).dt.month

In [206]:
# Dados de 2018

mes_2018 = dados_2018.loc[:,['mes','pessoas','ilesos','feridos_graves']]
k = mes_2018.groupby(by='mes').sum()
prop_ilesos = (k['ilesos']/k['pessoas'])
prop_feridos = (k['feridos_graves']/k['pessoas'])
prop = pd.DataFrame(prop_ilesos).rename(columns = {0:'Proporção de ilesos'}).round(3)
prop['Proporção de feridos graves'] = pd.DataFrame(prop_feridos).round(3)
prop

Proporção de ilesos  Proporção de feridos graves
mes                                                  
1                  0.513                        0.091
2                  0.498                        0.092
3                  0.482                        0.102
4                  0.430                        0.113
5                  0.422                        0.114
6                  0.425                        0.116
7                  0.438                        0.110
8                  0.427                        0.106
9                  0.421                        0.115
10                 0.430                        0.113
11                 0.433                        0.115
12                 0.433                        0.108

In [207]:
# Dados de 2019

dados_2019['mes'] = pd.to_datetime(dados_2019['data_inversa']).dt.month

In [208]:
# Dados de 2019

mes_2019 = dados_2019.loc[:,['mes','pessoas','ilesos','feridos_graves']]
k = mes_2019.groupby(by='mes').sum()
prop_ilesos = (k['ilesos']/k['pessoas'])
prop_feridos = (k['feridos_graves']/k['pessoas'])
prop = pd.DataFrame(prop_ilesos).rename(columns = {0:'Proporção de ilesos'}).round(3)
prop['Proporção de feridos graves'] = pd.DataFrame(prop_feridos).round(3)
prop

Proporção de ilesos  Proporção de feridos graves
mes                                                  
1                  0.424                        0.113
2                  0.424                        0.107
3                  0.433                        0.115
4                  0.416                        0.106
5                  0.415                        0.121
6                  0.424                        0.122
7                  0.437                        0.117
8                  0.413                        0.117
9                  0.421                        0.111
10                 0.419                        0.120
11                 0.426                        0.111
12                 0.422                        0.114

In [209]:
# Dados de 2020

dados_2020['mes'] = pd.to_datetime(dados_2020['data_inversa']).dt.month

In [210]:
# Dados de 2020

mes_2020 = dados_2020.loc[:,['mes','pessoas','ilesos','feridos_graves']]
k = mes_2020.groupby(by='mes').sum()
prop_ilesos = (k['ilesos']/k['pessoas'])
prop_feridos = (k['feridos_graves']/k['pessoas'])
prop = pd.DataFrame(prop_ilesos).rename(columns = {0:'Proporção de ilesos'}).round(3)
prop['Proporção de feridos graves'] = pd.DataFrame(prop_feridos).round(3)
prop

Proporção de ilesos  Proporção de feridos graves
mes                                                  
1                  0.416                        0.107
2                  0.420                        0.111
3                  0.401                        0.113
4                  0.376                        0.118
5                  0.401                        0.116
6                  0.423                        0.116
7                  0.424                        0.109
8                  0.423                        0.119
9                  0.416                        0.117
10                 0.411                        0.120
11                 0.407                        0.120
12                 0.412                        0.122

In [211]:
# Dados de 2021

dados_2021['mes'] = pd.to_datetime(dados_2021['data_inversa']).dt.month

In [212]:
# Dados de 2021

mes_2021 = dados_2021.loc[:,['mes','pessoas','ilesos','feridos_graves']]
k = mes_2021.groupby(by='mes').sum()
prop_ilesos = (k['ilesos']/k['pessoas'])
prop_feridos = (k['feridos_graves']/k['pessoas'])
prop = pd.DataFrame(prop_ilesos).rename(columns = {0:'Proporção de ilesos'}).round(3)
prop['Proporção de feridos graves'] = pd.DataFrame(prop_feridos).round(3)
prop

Proporção de ilesos  Proporção de feridos graves
mes                                                  
1                  0.426                        0.112
2                  0.414                        0.116
3                  0.411                        0.115
4                  0.406                        0.123
5                  0.416                        0.125
6                  0.431                        0.109
7                  0.432                        0.113
8                  0.421                        0.119
9                  0.417                        0.122
10                 0.414                        0.118
11                 0.427                        0.121
12                 0.421                        0.112

- Bom, como usei a mesma lógica para todos os anos, nesse caso, logo de início, para que eu pudesse resolver esse problema, importei a biblioteca "datetime" , assim usei a função "datatime()" juntamente com a função "dt.month", para pegar todas as datas e assim separa-las em meses. Com isso, usei também a função "groupby" para somar todos os meses de cada ano e depois criei a proporção que a questão estava pedindo, tanto para ilesos como para os feridos graves e finalizando, usei a função "round(3), para arredondar todos os valores para 3 casas decimais.

<div class="alert alert-block alert-info">
4º Tarefa - C
</div>

In [213]:
# Dados antes da pandemia
antes_da_pandemia = len(pd.concat([dados_2018,dados_2019]))

# Dados durante a pandemia
durante_a_pandemia = len(pd.concat([dados_2020,dados_2021]))

# Media dos anos
antes = antes_da_pandemia/2
durante = durante_a_pandemia/2

# Diferença das médias
diferença_pandemia = antes - durante

In [214]:
df_pandemia = pd.DataFrame([antes,durante],index = ['Antes da Pandemia','Depois da Pandemia']).rename(columns = {0:'Media de incidência de acidentes '})
df_pandemia

Media de incidência de acidentes 
Antes da Pandemia                             68370.5
Depois da Pandemia                            64031.5

In [215]:
diferenca = pd.DataFrame([diferença_pandemia],index = ['Diferença entre os períodos']).rename(columns = {0:"Valor da diferença"})
diferenca

Valor da diferença
Diferença entre os períodos              4339.0

Com isso, é perceptível que, houve uma dimunuição da incidência de acidentes nas rodovias federais por causa da pandemia.

- Nessa situação, basicamente eu juntei os anos de 2018 e 2019 usando a função "concat", pois esses dois anos foram os que não tinham pandemia ainda, e depois juntei os anos de 2020 e 2021 também usando a função "concat", pois esses dois anos foram os que passaram pela pandemia, logo em seguida, como tinha valores de dois anos nas duas situações eu fiz a media de cada o dos dois, pois os diferenciei com as variaveis "antes_da_pandemia" e "durante_a_pandemia", e depois fiz o cálculo da diferença entre eles, e com a função "rename" mudei os nomes tanto das linhas como da coluna.